### PR TITLE
contrib/minetest: new package (5.8.0)

### PR DESCRIPTION
**Pull request by ttyyls**

### CODE REVIEW

**Comment by ttyyls** [comment]
I've made an attempt at providing a service file with a _minetest user and tmpfile entry [here](https://github.com/chimera-linux/cports/compare/master...ttyyls:cports:minetest) bu I rolled it back since it'd require manual installation and loading of individual games and then changing the service file to make them work.

File: contrib/minetest/template.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+]
+sha256 = [
+    "610c85a24d77acdc3043a69d777bed9e6c00169406ca09df22ad490fe0d68c0c",
+    "2fde8e27144988210b9c0ff1e202905834d9d25aaa63ce452763fd7171096adc",
+]
+tool_flags = {"CXXFLAGS": ["-Wno-deprecated-declarations"]}
+hardening = ["!int"]
+
+
+def post_extract(self):
```
Comment (JamiKettunen) [inline contrib/minetest/template.py:48]: Drop whole `post_extract()` and `build_wrksrc` above and just use
```py
source_paths = [".", "lib/irrlichtmt"]
```

Comment (ttyyls) [reply contrib/minetest/template.py]: looks cleaner that way. thanks

File: contrib/minetest/template.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    self.pkgdesc = f"{pkgdesc} (common files)"
+    self.install_if = [f"{pkgname}={pkgver}-r{pkgrel}"]
+
+    return ["usr/share/minetest"]
+
+
+@subpackage("minetest-server")
+def _server(self):
+    self.pkgdesc = f"{pkgdesc} (server)"
+    self.depends = ["minetest-common"]
```
Comment (q66) [inline contrib/minetest/template.py:67]: needs to be versioned

Comment (q66) [reply contrib/minetest/template.py:67]: also probably main package should also depend on it, no?

Comment (ttyyls) [reply contrib/minetest/template.py]: >main package should also depend on it
yes and it does through the install_if on minetest-common

Comment (q66) [reply contrib/minetest/template.py]: does that make sense over just a hard dependency in the other direction though

Comment (ttyyls) [reply contrib/minetest/template.py]: i changed it into a hard dependency in the main package.

**Pull request merged**